### PR TITLE
Implement PHP strict types.

### PR DESCRIPTION
**Pull request by samuelwilliams**
Complete implementation of PHP 7 strict types as per Issue #8

### CODE REVIEW

File: src/IPBlock.php [hunk 1]
```diff
@@ -53,7 +55,7 @@ abstract class IPBlock implements \ArrayAccess, \IteratorAggregate, \Countable
     protected $nb_addresses;
 
     /**
-     * @var string Either "IPv4" or "IPv6"
+     * @var string|IP
```
Comment (rlanvin) [inline src/IPBlock.php]: This is always a string I think, never an IP object.

File: src/IP.php [hunk 2]
```diff
@@ -67,6 +69,11 @@ abstract class IP
      */
     protected $is_private;
 
+    /**
+     * @var array
+     */
+    protected static $private_ranges;
```
Comment (rlanvin) [inline src/IP.php]: I don't think this code should be in this PR.

File: src/IPBlockIterator.php [hunk 3]
```diff
@@ -33,7 +35,7 @@ class IPBlockIterator implements \Iterator, \Countable
     protected $first_block = null;
 
     /**
-     * @var int
+     * @var \GMP
```
Comment (rlanvin) [inline src/IPBlockIterator.php]: If this is a GMP instance, does it make sense to initialize it to "0"?

**Comment by rlanvin** [comment]
Thank you! I added a few comments, but otherwise it's looking good. Please also check the conflict so I can merge this.

File: src/IP.php (hunk 2 above)
Comment (samuelwilliams) [reply src/IP.php]: I agree, but it still needs to be in the code base.

File: src/IPBlockIterator.php (hunk 3 above)
Comment (samuelwilliams) [reply src/IPBlockIterator.php]: Agreed

File: src/IPBlock.php (hunk 1 above)
Comment (samuelwilliams) [reply src/IPBlock.php]: Agreed.

**Comment by rlanvin** [comment]
Thank you!

**Pull request merged**